keel-artifact/src/main/kotlin/com/netflix/spinnaker/keel/artifacts/ArtifactListener.kt

```diff
@@ -75,6 +77,44 @@ class ArtifactListener(
       }
   }
 
+  /**
+   * Finds the delivery configs that are using an artifact,
+   * and publishes a build lifecycle event for them.
+   */
+  fun createBuildLifecycleEvent(artifact: PublishedArtifact) {
```

Comment:
Maybe `publishBuildNotStartedEvent`, since this is the only event this function will ever publish? Also, why `NOT_STARTED` here since this is a new artifact version and so a build has definitely already started? I understand that sometimes we get the artifact event _before_ the build finished, but it's definitely at least running by the time we get here, right?

Reply:
BTW, `BuildMetadata` has a `status` field that we get from CI.

Reply:
So, right now, in order to kick of monitoring the event has to have a "NOT_STARTED" status. Then the build monitor updates to the correct status almost immediately. I'm going to change that in a different pr (to have a `monitor: Boolean` flag), but I don't want to wrap that change into this pr. But, I didn't know that I'd want to use these things this way until I had another type of thing to monitor! Fun discoveries using things that are freshly built :) 

In light of that,  I don't want to change the method name.

Reply:
On naming, maybe just `publishBuildLifecycleEvent` as a compromise, then? At least you use the same verb as in other places.

On the `NOT_STARTED` status, I think what's confusing here is that you seem to be overloading that status field to capture state for both the act of monitoring the build (where this status would totally make sense), and the status of the build itself. I think these should be separate fields. Maybe that's what you meant by the `monitor: Boolean` flag, not sure.

Reply:
@luispollo yep, my plan is to decouple the status from whether we have to monitor or not so that the status of the event can be the status of the actual think. I'll do that in an immediately following pr. I'll address naming changes in that PR because I'll need to touch the method anyway.